Add actual documentation to the readme.md (#87)
govau/chameleon: README.md

> Interactive colour template previews for the Design System.

> Chameleon allows users to change and share different colour palettes using the Australian Government Design System. It is an express server that takes query parameters altering the apperance of the template. 

Reviewer: Not feeling the express server:

- `express.js` - link to it
More like "Takes query params, modifies a html template and sends back html response"

> Chameleon allows users to change and share different colour palettes using the Australian Government Design System. It is an express server that takes query parameters altering the apperance of the template. 

Author: ```suggestion
Chameleon allows users to change and share different colour palettes using the Australian Government Design System. It is a server that takes query parameters altering the appearance of the template. 
```

You can access it through a user interface at [https://designsystem.gov.au/templates/home/customise](https://designsystem.gov.au/templates/home/customise) if you want to use it locally you can follow the instructions below. 

## Install

Clone chameleon into a local directory:
```
git clone git@github.com:govau/chameleon.git
```

Make sure you are in your cloned directory and install the dependencies:

```node
npm install
```

> Then run `npm start` or `npm watch`.

Reviewer: `npm run watch`

> Then run `npm start` or `npm watch`.

Reviewer: ```suggestion
Then run `npm run start` or `npm run watch`.
```

## Getting started

Once the server is running you should be able to connect to it with [http://localhost:3000/chameleon](http://localhost:3000/chameleon).

Chameleon looks like a full page template using the Australian Government Design System. It has multiple templates in the `/templates` directory. You can access the templates by navigating to the url [http://localhost:3000/chameleon/content](http://localhost:3000/chameleon/content).

To change the colours you can add a query parameter. [http://localhost:3000/chameleon/content?action=red&background=#eee](http://localhost:3000/chameleon/content?action=red&background=#eee).

> The supported query parameters are `text`, `action`, `focus`, `background` and `textDark`, `actionDark`, `focusDark`, `backgroundDark`. Using these queries you can modify the templates and share them.

Reviewer: Maybe make this more of a table or list looking thing? Not a big deal...

> The supported query parameters are `text`, `action`, `focus`, `background` and `textDark`, `actionDark`, `focusDark`, `backgroundDark`. Using these queries you can modify the templates and share them.

Author: ```suggestion
Using query parameters you can modify the templates and share them. The supported query parameters are:
 - `text`, the default text colour
 - `action`, the interactive colour used on buttons and links
 - `focus`, the focus colour when interactive elements receive focus
 - `background`, the background colour
 - `textDark`, dark variation of the text colour
 - `actionDark`, dark variation of the action colour
 - `focusDark`, dark variation of the focus colour
 - `backgroundDark`, dark variation of the background colour
```

> The supported query parameters are `text`, `action`, `focus`, `background` and `textDark`, `actionDark`, `focusDark`, `backgroundDark`. Using these queries you can modify the templates and share them.

Author: ```suggestion
Using query parameters you can modify the templates and share them. The supported query parameters are:
- `text`
- `action`
- `focus`
- `background`
- `textDark`
- `actionDark`
- `focusDark`
- `backgroundDark`
```

## How it works

> The express server reads the data from the HTML files based on the URL. If there are valid query parameters it generates a new stylesheet and replaces the `<link rel="stylesheet" href="/chameleon/assets/css/main.css">` with the new styles. If there is an invalid colour entered it will log errors in the page template returned to the user.

Reviewer: - `express.js`

"Reads a HTML file based on the URL, given valid query paramaters"

> The express server reads the data from the HTML files based on the URL. If there are valid query parameters it generates a new stylesheet and replaces the `<link rel="stylesheet" href="/chameleon/assets/css/main.css">` with the new styles. If there is an invalid colour entered it will log errors in the page template returned to the user.

Author: ```suggestion
The server ( [expressjs](https://expressjs.com/ ) reads the data from the HTML files based on the URL. If there are valid query parameters it generates a new stylesheet and replaces the `<link rel="stylesheet" href="/chameleon/assets/css/main.css">` with the new styles. If there is an invalid colour entered it will log errors in the page template returned to the user.
```
